File: scripts/fix_imports.py
import os
import re

ROOT_DIR = r"c:\yoburaj\billing_app_frontend"
APP_DIR = os.path.join(ROOT_DIR, "app")
SRC_DIR = os.path.join(ROOT_DIR, "src")

# The root level folders in src/
known_folders = ["components", "constants", "context", "database", "hooks", "services", "utils", "screens", "navigation"]

def update_imports(file_path):
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Simple regex to catch common relative up-path imports pointing to our root folders, including db
    # We replace from '../db/' to '@/database/', '../../components/' to '@/components/', etc.
    # We'll use a regex that looks for from ['"](\.\./)+([^/'"]+)
    
    def repl(match):
        prefix = match.group(1)
        folder = match.group(3)
        rest = match.group(4)
        
        # map old 'db' to 'database'
        if folder == 'db':
            folder = 'database'
            
        if folder in known_folders:
            return f"{prefix}@/{folder}{rest}"
        else:
            # return original if not matched
            return match.group(0)
            
    # Matches: import ... from '../../constants/Vegetables';
    # Group 1: string prefix (" or ')
    # Group 2: (../) repeated
    # Group 3: first folder after ../ (e.g., constants)
    # Group 4: rest of the path until quote
    # group 5: closing quote
    new_content = re.sub(r'([\'"])((?:\.\./)+)([^/\'"]+)(/[^\'"]*)[\'"]', repl, content)
    
    # What about sibling imports in src? like in src/components/A.tsx importing src/utils/B.ts
    # It might be `import B from '../utils/B'` which is already matched.
    
    if new_content != content:
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f"Updated imports in {file_path}")

for walk_dir in [APP_DIR, SRC_DIR]:
    if not os.path.exists(walk_dir):
        continue
    for root, dirs, files in os.walk(walk_dir):
        for file in files:
            if file.endswith('.ts') or file.endswith('.tsx'):
                update_imports(os.path.join(root, file))

print("Import references fixed!")

File: scripts/fix_imports.js
const fs = require('fs');
const path = require('path');

const ROOT_DIR = 'C:/yoburaj/billing_app_frontend';
const APP_DIR = path.join(ROOT_DIR, 'app');
const SRC_DIR = path.join(ROOT_DIR, 'src');

const knownFolders = ["components", "constants", "context", "database", "db", "hooks", "services", "utils", "screens", "navigation"];

function processFile(filePath) {
    const content = fs.readFileSync(filePath, 'utf8');

    // Regex matches: 'import ... from "../components/..."' or '"../../db/..."'
    const updatedContent = content.replace(/(['"])((?:\.\.\/)+)([^/'"]+)(\/[^'"]*)(['"])/g, (match, p1, p2, folder, rest, p5) => {
        if (folder === 'db') {
            folder = 'database';
        }
        if (knownFolders.includes(folder)) {
            return `${p1}@/${folder}${rest}${p5}`;
        }
        return match;
    }).replace(/(['"])(\.\/)([^/'"]+)(\/[^'"]*)(['"])/g, (match, p1, p2, folder, rest, p5) => {
        // matches `./db/` or `./utils/` in root level or something... wait, `./` is tricky.
        if (folder === 'db' && filePath.includes(path.join(ROOT_DIR, 'src')) && path.dirname(filePath) === SRC_DIR) {
            folder = 'database';
        }
        return match;
    });

    if (content !== updatedContent) {
        fs.writeFileSync(filePath, updatedContent, 'utf8');
        console.log(`Updated: ${filePath}`);
    }
}

function walk(dir) {
    if (!fs.existsSync(dir)) return;
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            walk(fullPath);
        } else if (fullPath.endsWith('.ts') || fullPath.endsWith('.tsx')) {
            processFile(fullPath);
        }
    }
}

walk(APP_DIR);
walk(SRC_DIR);
console.log("Done");
